### PR TITLE
symlinks GB-638

### PR DESCRIPTION
**Pull request by ngalaiko**

### CODE REVIEW

File: packages/tauri/src/gb_repository/repository.rs [hunk 1]
```diff
@@ -802,16 +802,24 @@ fn add_wd_path(
...
     // insert a pointer as the blob content instead
     // TODO: size limit should be configurable
-    let blob = if metadata.len() > 100_000_000 {
+    let blob = if metadata.is_symlink() {
+        // it's a symlink, make the content the path of the link
+        let link_target = std::fs::read_link(&file_path)?;
+        // if the link target is inside the project repository, make it relative
+        let link_target = link_target.strip_prefix(dir).unwrap_or(&link_target);
+        // bytes dance
+        let path_str = link_target.to_str().unwrap();
```
Comment (extrawurst) [inline packages/tauri/src/gb_repository/repository.rs]: why is it safe to `unwrap`?

File: packages/tauri/src/gb_repository/repository.rs [hunk 2]
```diff
@@ -802,16 +802,24 @@ fn add_wd_path(
...
     let modify_time = FileTime::from_last_modification_time(&metadata);
     let create_time = FileTime::from_creation_time(&metadata).unwrap_or(modify_time);
 
     // look for files that are bigger than 4GB, which are not supported by git
     // insert a pointer as the blob content instead
     // TODO: size limit should be configurable
-    let blob = if metadata.len() > 100_000_000 {
+    let blob = if metadata.is_symlink() {
+        // it's a symlink, make the content the path of the link
+        let link_target = std::fs::read_link(&file_path)?;
```
Comment (extrawurst) [inline packages/tauri/src/gb_repository/repository.rs:814]: is it possible that `link_target` is another smylink? does this have to be recursive?

Comment (ngalaiko) [reply packages/tauri/src/gb_repository/repository.rs:814]: [ede64d3](https://github.com/gitbutlerapp/gitbutler-client/pull/1494/commits/ede64d3501460432afa1b5efdb8cc0e3c5c49eb7) looks like it's ok!

File: packages/tauri/src/gb_repository/repository.rs (hunk 1 above)
Comment (ngalaiko) [reply packages/tauri/src/gb_repository/repository.rs]: i just copied it... removed it now! [3ebd071](https://github.com/gitbutlerapp/gitbutler-client/pull/1494/commits/3ebd071e9e8eb1fb364dfb03b8cd6bc7e8113127)

**Pull request merged**